lmp-device-auto-register: Improving User Guide (#271)
foundriesio/docs: source/user-guide/lmp-device-auto-register/lmp-device-auto-register.rst

.. _ug-lmp-device-auto-register:

Auto Register
=============

This section shows how to enable the ``lmp-device-auto-register`` recipe. This 
recipe creates a systemd oneshot service that will automatically register the 
device on first boot once it has internet connectivity.
This is done by providing an API Token that has **devices:create** scope.

.. warning::
   Do not use this recipe in production. This recipe is only intended for
   usage in a development environment. For more information, 

>    read :ref:`ref-factory-registration-ref`.

Reviewer: should we add that for production there is a more secure way to achieve the auto registration and that the customers should contact support in this case?

Reviewer 2: i would say to not use an API *key* for production. The recipe can still work as-is with the factory-registration-reference server.

The recipe lmp-device-auto-register_ is provided by meta-lmp_ and can be added by 
customizing your ``meta-subscriber-overrides.git``.

Prerequisites
-------------

To follow this section, it is important to have:

- Completed the :ref:`sec-learn` until the :ref:`gs-flash-device` section.

Instructions

> ------------

Reviewer 3: The same as the other PR. I think a reader coming to a UG wants more direct information, sometimes punctual information on how to perform one subtask.

I think we can drop the "Instructions" without changing the content.

Creating Token
^^^^^^^^^^^^^^

> In the right top corner, click on the avatar and select :guilabel:`Settings` in the 

Reviewer 3: I understand it's from the Foundries.io Dashboard, is that right?

Author: Made it much more clear now.

drop-down list.

.. figure:: /_static/git-config/settings.png
   :width: 900
   :align: center

   FoundriesFactory Settings

Select the tab :guilabel:`Tokens` and create a new **Api Token** by clicking on 
the :guilabel:`+ New Token`.
Complete with a **Description** and the **Expiration date** and select :guilabel:`next`.

Select the :guilabel:`device:create` token and 
select your **Factory**. You can later revoke this access and set up a new 
token once you are familiar with the :ref:`ref-api-access`.

.. figure:: /_static/userguide/lmp-device-auto-register/lmp-device-auto-register-token.png
   :width: 500
   :align: center

>    Token for source code access

Reviewer 2: ```suggestion
   Token for device creation access
```

      
    host:~$ gedit recipes-support/lmp-device-auto-register/lmp-device-auto-register/api-token

**recipes-support/lmp-device-auto-register/lmp-device-auto-register/api-token**:
      
.. prompt:: text
 
    <YOUR_API_TOKEN>

Create the file ``lmp-device-auto-register.bbappend`` in order to give the recipe 
access to the ``api-token`` file.

.. prompt:: bash host:~$, auto
      
    host:~$ gedit recipes-support/lmp-device-auto-register/lmp-device-auto-register.bbappend

**recipes-support/lmp-device-auto-register/lmp-device-auto-register.bbappend**:
      
.. prompt:: text
 
    FILESEXTRAPATHS_prepend := "${THISDIR}/${PN}:"

Add the changed files, commit and push:

      .. prompt:: bash host:~$, auto

          host:~$ git add recipes-samples/images/lmp-factory-image.bb
          host:~$ git add recipes-support/lmp-device-auto-register/lmp-device-auto-register/api-token
          host:~$ git add recipes-support/lmp-device-auto-register/lmp-device-auto-register.bbappend
          host:~$ git commit -m "lmp-device-auto-register: Adding recipe"
          host:~$ git push

The latest **Target** named ``platform-devel`` should be the CI job you just created.

> When FoundriesFactory CI finishes all jobs, download and flashes the image.

Reviewer: ```suggestion
When FoundriesFactory CI finishes all jobs, download and flash the image.
```

Testing Auto Register
^^^^^^^^^^^^^^^^^^^^^

> After boot the new image, if your device is connected to the internet, the device 

Reviewer: ```suggestion
After booting the new image, if your device is connected to the internet, the device 
```

> will automatically registerer and should be visible by navigating to the web 

Reviewer: ```suggestion
will automatically register to your factory and should be visible by navigating to the web 
```

selecting the **Devices** tab.

.. figure:: /_static/userguide/lmp-device-auto-register/lmp-device-auto-register-device.png
   :width: 900
   :align: center

   FoundriesFactory Device Auto Registered

On your device, use the following command to list the ``lmp-device-auto-register`` 
service:

.. prompt:: bash device:~$

    systemctl list-unit-files | grep enabled | grep lmp-device-auto-register

**Example Output**:

.. prompt:: text

    lmp-device-auto-register.service           enabled         enabled

Verify the ``lmp-device-auto-register`` application status:

.. prompt:: bash device:~$, auto

    device:~$  systemctl status lmp-device-auto-register

**Example Output**:

.. prompt:: text

     lmp-device-auto-register.service - Script to auto-register device into Factory
          Loaded: loaded (/usr/lib/systemd/system/lmp-device-auto-register.service; enabled; vendor preset: enabled)
          Active: active (exited) since Sun 2021-09-12 17:34:06 UTC; 5min ago
         Process: 774 ExecStart=/usr/bin/lmp-device-auto-register (code=exited, status=0/SUCCESS)
        Main PID: 774 (code=exited, status=0/SUCCESS)

Summary
^^^^^^^

> This user guide gives you an example of how to enable the LmP Device Auto Registered. 

Reviewer 3: The User Guide is the collection of several chapters, this page is one chapter.

I would drop the summary for the user guide entries, as I think the purpose of UG is to be more direct and punctual.

Author: I accepted your suggestions on removing the prerequisites, summary, and instructions.
